### PR TITLE
132 frontend add volunteer button to events detail page needing volunteers

### PR DESCRIPTION
**Pull request by natalier97**
- changed rsvp button name to 'attending'
- added conditional rendering to show 'attend' option if event _needs_ attendees
- updated volunteer application & added useStates --> ready to send information to backend

### CODE REVIEW

File: front-end/src/components/VolunteerApplication.jsx [hunk 1]
```diff
@@ -1,29 +1,89 @@
...
+    // } catch (error) {
+    //   console.log('Failed application submission', error);
+    // }
+  };
+
+  //handles checkbox selecting/deselecting, updates rolesSelected state accordingly
+  function handleClickRole(roleID) {
+
+    // find the role instance that matches the provided roleID
+    const roleInstance = roles.find((r) => r.id === roleID);
```
Comment (KaraWatts) [inline front-end/src/components/VolunteerApplication.jsx:59]: Is this just leftover from when you were pulling the whole instance?

File: front-end/src/components/VolunteerApplication.jsx [hunk 2]
```diff
@@ -1,29 +1,89 @@
...
 import Form from "react-bootstrap/Form";
 import Modal from "react-bootstrap/Modal";
+import EventDetails from "../pages/EventDetails";
+import { resolvePath } from "react-router-dom";
 
+//name, email, phone, best time to contact, have you volunteered with this organization/person before, why would you like to volunteer for this event, what skills/experience do you have? which role are you interested in?
 
+function VolunteerApplication({ show, handleClose, eventID }) {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
```
Comment (KaraWatts) [inline front-end/src/components/VolunteerApplication.jsx:14]: we should probably set the default here to the users account email - it should be stored in the user useContext still

File: front-end/src/utilities/EventUtilities.jsx [hunk 3]
```diff
@@ -134,5 +134,14 @@ export const timeZoneAbbreviations = [
...
   ];
+
+
+
+  //view volunteer roles for specific event
+export const volunteerRoles = async (eventID) => {
+  const response = await api.get(`events/${eventID}/volunteers/`);
+  let rolesArr = response.data;
+  return rolesArr;
+};
```
Comment (KaraWatts) [inline front-end/src/utilities/EventUtilities.jsx:146]: this call is actually meant for the admin page to give all of the role data the event details will have all of the role data that you need for the volunteer application.  This is great to have here for the admin page though!

**Review by KaraWatts — CHANGES_REQUESTED** [review]
Looks great, just a couple things and a question.

File: front-end/src/components/VolunteerApplication.jsx (hunk 1 above)
Comment (natalier97) [reply front-end/src/components/VolunteerApplication.jsx:59]: ahh yes, i'll update it now.

File: front-end/src/utilities/EventUtilities.jsx (hunk 3 above)
Comment (natalier97) [reply front-end/src/utilities/EventUtilities.jsx:146]: i'm going to keep it like that for now until that's completed so we can see the roles for now. you can create an issue to update it when that's done.

File: front-end/src/components/VolunteerApplication.jsx (hunk 2 above)
Comment (natalier97) [reply front-end/src/components/VolunteerApplication.jsx:14]: that's a good idea, let me look at that if ican figure it out quickly

**Pull request merged**